### PR TITLE
Problem: Crash when trying to auth via websocket

### PR DESCRIPTION
**Pull request by olethanh**
The auth function in websocket was crashing when the header  "X-Auth-Signature" wasn't passed, even  that authentification method wasn't used.

Make it an option in the method so we can still have error raised in normal header auth

### CODE REVIEW

File: src/aleph/vm/orchestrator/views/__init__.py [hunk 1]
```diff
@@ -337,11 +337,11 @@ async def status_public_config(request: web.Request):
     )
 
 
-def authenticate_api_request(request: web.Request) -> bool:
+def authenticate_api_request(request: web.Request, raises_on_missing_header=True) -> bool:
```
Comment (hoh) [inline src/aleph/vm/orchestrator/views/__init__.py:340]: What about making this the default behavior and skipping the extra argument ?

Comment (olethanh) [reply src/aleph/vm/orchestrator/views/__init__.py:340]: The update allocation endpoint rely on that behaviour

**Comment by hoh** [comment]
Calling `authenticate_api_request` from a websocket endpoint does not make sense, and it seems more appropriate to now allow header authentication on that endpoint.

My proposal on this branch: https://github.com/aleph-im/aleph-vm/pull/632

**Review by hoh — CHANGES_REQUESTED** [review]
My proposal on this branch: https://github.com/aleph-im/aleph-vm/pull/632

**Pull request merged**